lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -1,49 +1,91 @@
-            $("head").append(linkDemo)
-
-            # Adding DOM element to place formula into
-            span = $("<span>").attr("id", "formula")
-            $("body").append(span)
-
-            require(["ace/ext/katex/katex"], (katexInner) ->
-                katex = katexInner
-                onLoaded()
-                return
-            )
-            return
-
-        onLoaded = ->
-            selectedText = editor.getSelectedText()
-            try
-                katex.render(selectedText, $("#formula")[0])
-            catch e
-                $("#formula").text(e.message)
-            return
-
-        callback = (editor) ->
-            unless katex?
-                initKaTeX(onLoaded)
-                return
-            onLoaded()
-
-        editor.commands.addCommand({
-            name: "previewLaTeXFormula",
-            bindKey: {win: "Alt-p", mac: "Alt-p"},
-            exec: callback
-        })
+  exports.setupPreviewer = (editor, popoverHandler) ->
+    katex = null
+    popoverHandler = popoverHandler ? {
+      show: (jqPopoverContainer, options) ->
+        jqPopoverContainer.css($("textarea.ace_text-input").position())
+        popoverPosition = jqPopoverContainer.position()
+        popoverPosition.top += 24
+        jqPopoverContainer.css(popoverPosition)
+        jqPopoverContainer.popover(options)
+        jqPopoverContainer.popover("show")
         return
+
+      hide: (jqPopoverContainer) ->
+        jqPopoverContainer.popover("destroy")
+
+      isVisible: (popoverElement) ->
+        popoverElement.children(".popover").is(":visible")
+    }
+
+    initKaTeX = (onLoaded) ->
+      # Adding CSS for demo formula
+      cssDemoPath = require.toUrl("./katex-demo.css")
+      linkDemo = $("<link>").attr(
+        rel: "stylesheet"
+        href: cssDemoPath
+      )
+      $("head").append(linkDemo)
+
+      # Adding DOM element to place formula into
+      a = $("<a>").attr(
+        href: "#"
+        id: "formula"
+        "data-toggle": "popover"
+      )
+      $("body").append(a)
+
+      require(["ace/ext/katex"], (katexInner) ->
+        katex = katexInner
+        onLoaded()
+        return
+      )
+      return
+
+    callbackHidePopover = () ->
+      popoverHandler.hide($("#formula"))
+      editor.off("changeSelection", callbackHidePopover)
+      editor.session.off("changeScrollTop", callbackHidePopover)
+      editor.session.off("changeScrollLeft", callbackHidePopover)
+      return
+
+    onLoaded = ->
+      options = {
+        html: true
+        placement: "bottom"
+        trigger: "manual"
+        title: "Formula"
+        container: "#editor"
+      }
+      try
+        options.content = katex.renderToString(
+          editor.getSelectedText(),
+          {displayMode: true}
+        )
+      catch e
+        options.content = e
+      finally
+        popoverHandler.show($("#formula"), options)
+        editor.on("changeSelection", callbackHidePopover)
```

Comment:
You'll want to add a namespace suffix (see http://api.jquery.com/on/ ) to avoid conflicts with other handlers, and call `off()` before `on()`, otherwise you may have troubles. See e.g. this jsfiddle: https://jsfiddle.net/6vcjmzbj/ where every click adds a new handler to the button.